gce/survival-training/README.md

```diff
@@ -0,0 +1,93 @@
+# GCE survival training
+
+Training jobs that survive GCE shutdowns
+
+- - -
+
+[Google Compute Engine](https://cloud.google.com/compute/docs/) (GCE) gives you direct access to the computational backbone of Google Cloud Platform. You can use GCE to create virtual machines provisioned with the resources of your choosing ([satisfying the constraints of any quotas that you define](https://cloud.google.com/compute/quotas)).
+
+This has particular value when it comes to training machine learning models, as this gives you access to resources beyond those available to you physically. For example, GCE will soon offer preemptible GPU instances. This means that, if you designed your trainer appropriately, your cost per training step could be a fraction of what it would be on a dedicated instance.
+
+Even if you want to train on a dedicated instance, GPU-enabled GCE instances have always had the issue that they go down about once a week for maintenance. This introduces a significant amount of overhead (in both time and money) to performing training on GCE instances.
+
+However, GCE *does* expose primitives that allow you set up training jobs and your GCE instances in a manner that lets you train across instance preemptions and shutdowns.
+
+This guide provides you with a template to do exactly that.
+
+
+## Survival strategy
+
+There are five components to our resilient training jobs:
+
+1. [GCE custom images](https://cloud.google.com/compute/docs/images#custom_images)
+
+1. A trainer command line interface
+
+1. [GCE startup scripts](https://cloud.google.com/compute/docs/startupscript)
+
+1. [GCE instance metadata](https://cloud.google.com/compute/docs/storing-retrieving-metadata)
+
+1. [Cloud Source Repositories](https://cloud.google.com/source-repositories/)
```

Comment:
same comment as on the other README file.  is this necessary?  it somewhat adds additional complexity to the solution.  why not just pull from github, or store the code on GCS?

Reply:
This step is more meaningful for code that doesn't already conform to the interface we expect in the startup script. In general, and *unlike* the case with the cifar10 estimator, more work will be required on the part of the user to deploy their training job in a resilient manner to GCE.

Reply:
would it work similarly to pull from github as another alternative?

Reply:
Yep, could totally pull from GitHub - as long as login is not required. Nice thing about gcloud source repos is that you don't have to authenticate at all. For tensorflow/models, this shouldn't be an issue, but I didn't want to set a false expectation in the general case.

@amygdala , @dizcology , @puneith : Do you think I should modify the startup script corresponding to this guide to clone from GitHub?